src/System.Diagnostics.Process/src/System/Diagnostics/Process.Win32.cs

```diff
@@ -341,5 +342,98 @@ private bool WaitForInputIdleCore(int milliseconds)
             }
             return idle;
         }
+
+        /// <summary>Checks whether the argument is a direct child of this process.</summary>
+        /// <remarks>
+        /// A child process is a process which has this process's id as its parent process id and which started after this process did.
+        /// </remakrs>
+        private bool IsParentOf(Process possibleChildProcess) =>
+            StartTime < possibleChildProcess.StartTime
+            && Id == possibleChildProcess.GetParentProcessId();
+
+        /// <summary>
+        /// Attempts to get the process's parent process id.
+        /// </summary>
+        private unsafe int? GetParentProcessId()
+        {
+            // UNDONE: NtQueryInformationProcess will fail if we are not elevated and other process is. Advice is to change to use ToolHelp32 API's
```

Comment:
I would remove this comment - I thought @JeremyKuhne commented on that, maybe it got lost

Reply:
I mean make it something like
`NtQueryInformationProcess will fail if we are not elevated and other process is. Just return null and worst case we will not kill some children.`

Reply:
It's still on my to-do list, :-) There's that, along with adjusting the exception catch-alls, trying to get this working on OSX and maybe adding some caching of parent IDs (so that value isn't retrieved multiple times while the tree is crawled)

Reply:
Hm...the comment on line 359 came from msbuild. I'm wondering if the situation described can actually occur. 

If there was an elevation mis-match, wouldn't the open handle request have failed (because the requested handle permissions couldn't be granted), making this comment irrelevant?

Reply:
Only one way to know for sure...

Reply:
:-) Just checked. The request for the handle fails. Comment is irrelevant. Removed. 

The revised exception handling code will handle the handle request failure.

Reply:
OK good. By curious happenstance I believe I was the one that wrote all that code you are copying, so I would have added that comment, but it was nearly 15 years ago so I have no memory of any details.

Reply:
That's neat history!